android/src/main/java/io/jari/fingerprint/FingerprintModule.java

```diff
@@ -14,171 +13,125 @@
 import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 
-import java.util.HashMap;
-import java.util.Map;
+public final class FingerprintModule extends ReactContextBaseJavaModule {
 
-@SuppressWarnings("MissingPermission")
-public class FingerprintModule extends ReactContextBaseJavaModule {
-    FingerprintManager fingerprintManager;
-    ReactApplicationContext reactContext;
-    CancellationSignal cancellationSignal;
-    boolean isCancelled = false;
+    private static final int FINGERPRINT_ACQUIRED_AUTH_FAILED = 999;
+    private static boolean IS_CANCELED = false;
+    private static CancellationSignal CANCELLATION_SIGNAL;
 
-    final int FINGERPRINT_ACQUIRED_AUTH_FAILED = 999;
+    private final FingerprintManagerCompat fpm;
 
-    public FingerprintModule(ReactApplicationContext reactContext) {
-        super(reactContext);
-
-        this.reactContext = reactContext;
-        this.fingerprintManager = reactContext.getSystemService(FingerprintManager.class);
+    public FingerprintModule(final ReactApplicationContext rctx) {
+        super(rctx);
+        fpm = FingerprintManagerCompat.from(rctx);
     }
 
     @Override
-    public Map<String, Object> getConstants() {
```

Comment:
Why remove the constants?  
What if I want to handle a specific error from my JS?  
There has to be another solution rather than completely removing the constants...